### ISSUE TITLE
Indentation breakage caused by first character open parentheses string literal and apostrophe in comment

### ISSUE THREAD

**Issue by psanford** [dominikh/go-mode.el#282]
The following code does not indent properly after the return statement:

```package main

var templTrace = `
(
`

func main() {
	if isUserAnnotation {
		return // don't display unrelated user annotation events.
}
}
```

(This is a minimal example I winnowed down from go/src/cmd/trace/trace.go on the 1.12.7 tag.)

Removing the apostrophe from the comment or not having the open parentheses as the first character on the line in the string literal makes the buffer indent properly.

I've seen some other examples in the stdlib where indentation gets off after a comment with an apostrophe in it. I'll investigate some more to see they are also related to string literals or something else.

**Comment by dominikh** [comment]
Notice the two following things:

1. We're not indenting it incorrectly, we're not indenting it at all for some reason. You could prefix either of the two closing curly braces with any amount of whitespace, and we wouldn't touch it.
2. With show-paren-mode enabled, the two closing curly braces are shown as mismatched. Something about the raw string literal or the comment breaks Emacs's parser.

From what I remember, teaching Emacs about Go's raw string literals required some amount of hackery and custom parsing, because Emacs's own syntax table has no notion of a multi-line string literal that doesn't accept escape sequences. I believe some of that hackery is in the go-propertize-syntax function.

**Comment by psanford** [comment]
This was fixed by #283
